### PR TITLE
CBL-4519: Remove API_AVAILABLE annotation for macOS 10.12+ and iOS 10.0+

### PR DESCRIPTION
**Pull request by velicuvlad**
https://issues.couchbase.com/browse/CBL-4519

- removed API_AVAILABLE where version is less than the minimum deployment target

### CODE REVIEW

**Comment by velicuvlad** [comment]
blocked by https://github.com/couchbase/couchbase-lite-core/pull/1807 - will rebase when merged

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:169]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:171]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:193]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:195]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:207]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:209]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:219]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:221]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:324]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:326]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:398]: Remove. Test needs to do this instead.

File: Crypto/PublicKey+Apple.mm
Comment (pasin) [inline Crypto/PublicKey+Apple.mm:400]: Remove. Test needs to do this instead.

**Review by pasin — CHANGES_REQUESTED** [review]

**Comment by jianminzhao** [comment]
Approve contingent on Pasin's comments about gC4ExpectExceptions;
API_AVAILABLE is slick. Must be something with objective-c compiler.

**Review by pasin — APPROVED** [review]

**Pull request merged**
